### ISSUE TITLE
Opening parenthesis '(' after a function name in the Console crashes spyder

### ISSUE THREAD

**Issue by MathieuSch-101** [spyder-ide/spyder#15662]
### What steps will reproduce the problem?

I have enabled a bunch of options from the settings, and now I can not use the console anymore as it crashes each time I enter an opening parenthesis after a function name.

The exact settings which toggle on/off the crash is:
     Help -> Automatic connections -> IPython Console

For now, I have disabled this feature. 

I also have:
     IPython console -> Advanced settings -> Jedi completion: Enable
     IPython console -> Advanced settings -> Symbolic mathematics: Enable

For a complete configuration, please indicate to me which file I should provide.

Code that does not crash the console (and spyder):

     In [1]: (

Code which crashes the console (and spyder):

     In [1]: import numpy as np
     In [2]: np.abs(

The opening bracket is not actually displayed in the console, the crash occurs as soon as the key '(' is stroke.

Another example which crashes the console (and spyder):

     In [1]: max(

Versions:

    Python 3.9.1

    ipython 7.22.0
    jedi 0.17.2
    numpy 1.20.3
    mpmath 1.2.1
    spyder 5.0.3
    spyder-kernels 2.0.3
    sympy 1.8

**Comment by dalthviz** [comment]
Hi @MathieuSch-101 thanks for the feedback ! Did you get any traceback before Spyder crashes? Could you try to run spyder from the console/terminal with the `--debug-info verbose` flag to check for a traceback? Also, do you have Kite installed? Let us know :)

**Comment by MathieuSch-101** [comment]
@dalthviz No traceback at all for the crash, I do have Kite installed.
I will try to reproduce the issue with the debug mode; I had a different problem with another library and updated my python environment to 3.9.5 since then.

**Comment by dalthviz** [comment]
And with the Python update you still get the crash, right? Could you try to create a new env to do a clean installation with Python 3.9.5 and check again? Thanks!

**Comment by MathieuSch-101** [comment]
@dalthviz Bad news.. I could not reproduce the problem with python 3.9.5. I enabled as many features from the settings as I could, and the 'Help -> Automatic connections -> IPython Console' feature worked as expected. No crash now. 

I didn't think of the debug mode while I was on 3.9.1, my bad, and I guess this issue is now closed unless someone else experiences the same behavior. Although as this is not the default setting of spyder, I doubt many people will be testing this.
